src/msg/async/AsyncMessenger.cc

```diff
@@ -291,7 +291,15 @@ void AsyncMessenger::ready()
 {
   ldout(cct,10) << __func__ << " " << get_myaddr() << dendl;
 
-  stack->start();
+  stack->ready();
+  if (pending_bind) {
```

Comment:
@yuyuyu101 ,

As I said, I also preferred to have all the change inside RDMAStack.

But, there were two major problems I saw, with implementing according to that approach:
1. In Processor::start() the listener socket needs to supply a fd to to register  the event on. In RDMA-CM case, I need to start the librdmacm stack for that - and we don't want to do it before the fork. The only other option I can think of, is to have an intermediate eventfd that will be created before the fork. And after the fork, the real channel->fd event will write to the eventfd. 
This will open the door for other bugs - having 2 events for every listener event is not efficient nor good coding, the less events and contexts, the better we could sleep :)
here is a quick pseudo code to explain what I meant:
   1. RDMAServerSocket() { my_eventfd = eventfd(); }
   1. int RDMAServerSocket::fd() { return my_eventfd; };
   1. post_fork:
      1. channel = rdma_create_id(); // This will call ibv_open_device()
      1. create_file_event(channel->fd) { listen_handler; }
      1. RDMAServerSocket::listen_handler() { write(listen_handler, ...); }
1. If we call Processor::bind() before the fork, and postpone it inside RDMAStack(), which port should I return that Processor will save in Processor::listenaddr? Should I clone  the code with the logic of scanning for free ports from Processor::bind into RDMAStack?

This is why I implemented it like I did. It should have zero influence on other stack types, only for RDMAStack the logic of pending_listen should be used. And as you can see the change is minimal and very not risky.
Considering the options, I think it the best option we have.

Amir

@Adirl @orendu